File: App/src/main/java/uk/co/ianfield/devstat/widget/InformationPageFragment.java
package uk.co.ianfield.devstat.widget;

import android.content.ClipData;
import android.content.ClipboardManager;
import android.content.Context;
import android.os.Build;
import android.support.v4.app.Fragment;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.widget.Toast;

import org.androidannotations.annotations.AfterViews;
import org.androidannotations.annotations.EFragment;
import org.androidannotations.annotations.ViewById;

import java.util.ArrayList;

import uk.co.ianfield.devstat.R;
import uk.co.ianfield.devstat.StatItemAdapter;
import uk.co.ianfield.devstat.model.StatItem;

/**
 * Created by Ian Field on 14/08/15.
 */
@EFragment(R.layout.fragment_information_page)
public class InformationPageFragment extends Fragment {
    @ViewById(R.id.recycler_view)
    RecyclerView mRecyclerView;

    ArrayList<StatItem> items;

    @AfterViews
    void init() {
        setRetainInstance(true);
        mRecyclerView.setHasFixedSize(true);

        RecyclerView.LayoutManager layoutManager = new LinearLayoutManager(getActivity());
        mRecyclerView.setLayoutManager(layoutManager);

        RecyclerView.Adapter adapter = new StatItemAdapter(items, new StatItemAdapter.OnItemLongClickListener() {
            @Override
            public void onItemClick(int position) {

                if(Build.VERSION.SDK_INT < android.os.Build.VERSION_CODES.HONEYCOMB) {
                    android.text.ClipboardManager clipboard = (android.text.ClipboardManager) getActivity().getSystemService(Context.CLIPBOARD_SERVICE);
                    clipboard.setText(items.get(position).toString());
                } else {
                    ClipboardManager clipboard = (ClipboardManager) getActivity().getSystemService(Context.CLIPBOARD_SERVICE);
                    ClipData clip = android.content.ClipData.newPlainText("text label", items.get(position).toString());
                    clipboard.setPrimaryClip(clip);
                }
                Toast.makeText(getActivity(), R.string.copied_to_clipboard, Toast.LENGTH_SHORT).show();
            }
        });
        mRecyclerView.setAdapter(adapter);
    }

    public void setItems(ArrayList<StatItem> items) {
        this.items = items;
    }
}

File: app/src/main/java/uk/co/ianfield/devstat/widget/InformationPageFragment.kt
package uk.co.ianfield.devstat.widget

import android.content.ClipData
import android.content.ClipboardManager
import android.content.Context
import android.os.Bundle
import com.google.android.material.snackbar.Snackbar
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.LinearLayoutManager
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import uk.co.ianfield.devstat.R
import uk.co.ianfield.devstat.StatItemAdapter
import uk.co.ianfield.devstat.databinding.FragmentInformationPageBinding
import uk.co.ianfield.devstat.model.StatItem
import java.util.*

class InformationPageFragment : Fragment() {
    private var _binding: FragmentInformationPageBinding? = null
    private val binding get() = _binding!!

    private lateinit var items: ArrayList<StatItem>

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
        _binding = FragmentInformationPageBinding.inflate(inflater, container, false)
        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        binding.recyclerView.setHasFixedSize(true)
        val layoutManager = LinearLayoutManager(activity)
        binding.recyclerView.layoutManager = layoutManager

        val adapter = StatItemAdapter(requireContext(), items) { position: Int ->
            val clipboard = context?.getSystemService(Context.CLIPBOARD_SERVICE) as ClipboardManager
            val clip = ClipData.newPlainText("text label", items[position].toString())
            clipboard.setPrimaryClip(clip)
            Snackbar.make(binding.recyclerView, R.string.copied_to_clipboard, Snackbar.LENGTH_SHORT).show()
        }
        binding.recyclerView.adapter = adapter
    }

    fun setItems(items: ArrayList<StatItem>) {
        this.items = items
    }
}
